examples/basic/variables.tf

```diff
@@ -2,32 +2,35 @@
 # Input variables
 ########################################################################################################################
 
-variable "ibmcloud_api_key" {
+variable "resource_group" {
   type        = string
-  description = "The IBM Cloud API Key"
-  sensitive   = true
+  description = "An existing resource group name to use for this example. If not specified, a new resource group is created."
+  default     = null
 }
 
-variable "region" {
+variable "ibmcloud_api_key" {
+  description = "Used with the Terraform IBM-Cloud/ibm provider"
+  sensitive   = true
   type        = string
-  description = "Region to provision all resources created by this example"
-  default     = "us-south"
 }
 
-variable "prefix" {
+variable "region" {
+  default     = "us-south"
+  description = "Used with the Terraform IBM-Cloud/ibm provider as well as resource creation."
   type        = string
-  description = "Prefix to append to all resources created by this example"
-  default     = "basic"
+  validation {
+    condition     = contains(["eu-de", "eu-gb", "jp-tok", "us-south"], var.region)
+    error_message = "The IBM Cloud region to use must be one of: eu-de, eu-gb, jp-tok or us-south"
+  }
 }
 
-variable "resource_group" {
+variable "prefix" {
+  description = "The name to be used on all Watson resources as a prefix."
   type        = string
-  description = "The name of an existing resource group to provision resources in to. If not set a new resource group will be created using the prefix variable"
-  default     = null
-}
+  default     = "watsonx"
 
-variable "resource_tags" {
-  type        = list(string)
-  description = "Optional list of tags to be added to created resources"
-  default     = []
+  validation {
+    condition     = var.prefix != "" && length(var.prefix) <= 25
+    error_message = "You must provide a value for the prefix variable and the prefix length can't exceed 25 characters."
+  }
```

Comment:
I wouldn't add validation to the example - just add it to the module, otherwise you need to maintain in 2 places